bots/image-download

```diff
@@ -141,48 +139,87 @@ def download(link, force, quiet, stores):
         ret = curl.wait()
         if ret != 0:
             real_url = urlparse.urljoin("{0}://{1}:{2}{3}".format(url.scheme, sockaddr[0], sockaddr[1], url.path), name)
-            raise Exception("curl: unable to download %s (returned: %s)" % (real_url, ret))
+            raise RuntimeError("curl: unable to download %s (returned: %s)" % (real_url, ret))
 
         os.chmod(temp, stat.S_IRUSR | stat.S_IRGRP | stat.S_IROTH)
-        shutil.move(temp, dest)
+
+        # Due to time-cond the file size may be zero
+        # A new file downloaded, put it in place
+        if not exists or os.path.getsize(temp) > 0:
+            shutil.move(temp, dest)
+
     finally:
         # if we had an error and the temp file is left over, delete it
         if os.path.exists(temp):
             os.unlink(temp)
 
-    # Handle alternate TEST_DATA
+# Calcalate a place to put images where links are not committed in git
+def state_target(image):
```

Comment:
Maybe call the argument `path`? `image` is a bit confusing here. (Also, calc*a*late typo in the comment, but unimportant)

Reply:
We transform an image to a path here. I don't really care either way though. So changed.